sky/task.py

```diff
@@ -68,6 +68,45 @@ def _is_valid_env_var(name: str) -> bool:
     return bool(re.fullmatch(_VALID_ENV_VAR_REGEX, name))
 
 
+def _fill_in_env_vars_in_storage_config(
+        storage_config: Dict[str, Any], task_envs: Dict[str,
+                                                        str]) -> Dict[str, Any]:
+    """Detects env vars in storage_config and fills them with task.envs.
+
+    This func tries to replace two fields in storage_config: 'source' and
+    'name'.
+
+    Env vars of the following forms are detected:
+        - ${ENV}
+        - $ENV
+    where <ENV> must appear in task.envs.
+
+    Example:
+        {'mode': 'MOUNT', 'name': '${GSBUCKET}', 'store': 'gcs'}
+        -> {mode: 'MOUNT', name: 'mybucket', store: 'gcs'}
+    """
+
+    # TODO(zongheng): support ${ENV:-default}?
```

Comment:
I think it is ok to not support this, as the user can specify the default value for the env in the yaml file. We can keep this though.

Reply:
Sg. From user issue it seems that this syntax is applied in the run section. Keeping the TODO.